File: src/main/scala/de/sciss/kontur/desktop/impl/BasicPathField.scala
package de.sciss.kontur
package desktop
package impl

import swing.{Dialog, Component}
import java.io.File
import de.sciss.desktop.{Window, Preferences}

object BasicPathField {
  private final class Button(mode: PathField.Mode) extends PathButton(mode) {
    protected def showDialog(dialog: Dialog): Unit =
      Window.showDialog(Component.wrap(this), dialog)
  }
}
class BasicPathField(prefs: Preferences.Entry[File], default: File)(mode: PathField.Mode = PathField.Input)
  extends PrefPathField(prefs, default: File)(mode) {

	override protected def createPathButton(): PathButton  = new BasicPathField.Button(mode)
}

File: src/main/java/legacy/BasicPathField.java
package legacy;

import java.awt.Dialog;

public class BasicPathField
extends PrefPathField
{
	public BasicPathField( int type, String dlgTxt )
	{
		super( type, dlgTxt );
	}

	protected PathButton createPathButton( int buttonType )
	{
		return new Button( buttonType );
	}

	private static class Button
	extends PathButton
	{
		protected Button( int type )
		{
			super( type );
		}

		protected void showDialog( Dialog dlg )
		{
//			dlg.setVisible( true );
			BasicWindowHandler.showDialog( dlg );
		}
	}
}
